moviebot/nlu/neural_nlu.py

```diff
@@ -61,22 +61,32 @@ def generate_dacts(
             return selected_option
 
         intent, slots = self.annotate_utterance(user_utterance)
+        intent = UserIntents[intent]
 
         constraints = []
         operator = None
         for slot in slots:
-            if slot["slot"] == "MODIFIER":
+            if "MODIFIER" in slot["slot"]:
                 operator = self.get_constraint_operator(slot["value"])
                 continue
+
+            if intent == UserIntents.INQUIRE and "INQUIRE" not in slot["slot"]:
+                continue
+            if (
+                intent in (UserIntents.REMOVE_PREFERENCE, UserIntents.INQUIRE)
+                and "PREFERENCE" not in slot["slot"]
+            ):
+                continue
+            slot_name = Slots[slot["slot"].split("_")[1]].value
```

Comment:
To be checked but should you take the last part of the split? For example if the slot is `B_PREFERENCE_GENRES` then with the current implementation `PREFERENCE` would be used to find the slot name. 


```suggestion
            slot_name = Slots[slot["slot"].split("_")[-1]].value
```

Reply:
Good catch.